### PR TITLE
[BugFix] support all setQuantifier in special agg functions in new paser

### PR DESCRIPTION
**Pull request by packy92**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #21412

## Problem Summary(Required) ：

support all setQuantifier in special agg functions in new paser.  The ALL in agg fucntion like `count(all col)` does not affect the result, it just makes the new version compatible with the SQL that can be run on the old version.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [x] 2.4
  - [x] 2.3

### CODE REVIEW

File: fe/fe-core/src/test/java/com/starrocks/sql/parser/ParserTest.java [hunk 1]
```diff
@@ -302,6 +319,22 @@ private static Stream<Arguments> multipleStatements() {
         sqls.add(Pair.create("select 1 xxx select 2 xxx", false));
         return sqls.stream().map(e -> Arguments.of(e.first, e.second));
     }
+
+    private static Stream<Arguments> setQuantifierInAggFunc() {
```
Comment (imay) [inline fe/fe-core/src/test/java/com/starrocks/sql/parser/ParserTest.java:323]: Why not add cases in SQL-test which can get real results?

Comment (packy92) [reply fe/fe-core/src/test/java/com/starrocks/sql/parser/ParserTest.java:323]: We already have these integration tests in StarRocksSqllogicTest project. So I decide only add some FE UT.

**Review by kangkaisen — APPROVED** [review]

**Review by HangyuanLiu — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.4

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.3

**Pull request merged**